Assets/Plugins/Source/Core/RuntimeConfig.cs

```diff
@@ -0,0 +1,258 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+//#define EOS_RUNTIME_NEW_CONFIG_SYSTEM
+
+namespace PlayEveryWare.EpicOnlineServices
+{
+    using Epic.OnlineServices.Auth;
+    using Epic.OnlineServices.IntegratedPlatform;
+    using Epic.OnlineServices.Platform;
+    using System;
+
+#if EOS_RUNTIME_NEW_CONFIG_SYSTEM
+    /// <summary>
+    /// Contains the information necessary to configure the EOS SDK to connect
+    /// properly to Epic Online Services. Is ignorant of platform, because it
+    /// _only_ represents the _current_ runtime values to use. The values of the
+    /// field members defined within will need to be determined at runtime.
+    ///
+    /// _Project Settings_ is the area where there will be options to define
+    /// configuration values on a platform by platform basis. The EOS Plugin
+    /// allows users who are making their game with EOS to define different
+    /// properties for each platform, and to store sandbox / deployment
+    /// overrides so that it is easy to switch between deployment environments
+    /// during development. Those settings are distinct from these set of
+    /// values.
+    ///
+    /// Most of the values for these field members come from the
+    /// [Development Portal](https://dev.epicgames.com/portal/)
+    /// </summary>
+    public readonly struct RuntimeConfig
+    {
+
+        /*
+         * The following region contains values that are required in order to
+         * use the EOS SDK.
+         */
+        #region EOS SDK Configuration Values
+
+        /// <summary>
+        /// Product Name defined in the
+        /// 
+        /// </summary>
+        public readonly string ProductName;
+
+        /// <summary>
+        /// Version of Product.
+        /// </summary>
+        public readonly Version ProductVersion;
+
+        /// <summary>
+        /// Product Id
+        /// </summary>
+        public readonly Guid ProductId;
+
+        /// <summary>
+        /// Sandbox Id
+        /// </summary>
+        public readonly Guid SandboxId;
```

Comment:
Are sandbox ids always guids? I looked at the dev portal and saw that our sandbox id for dev is `p-pxx285jnqt8p3abnsekbcvhjwcgtk7` which is definitely not a guid. But then I looked at the sandbox for live `30f622bbe1f447889a26eb696d9fbb2a` and that's a guid.

Deployments on our product look like a GUID. The Product ID looks like a GUID.

Reply:
Good catch! It looks like if the sandbox is a "Private" sandbox, then it starts with a p, and is not itself a GUID. I wonder what it _is_.